### PR TITLE
Jakarta EE Entitiy and EJB support

### LINKED ISSUE #4827: Wizard "New Entity Classes from Database" does not work with Payara (version 16 RC1) - Jakarta EE 9.1
### Apache NetBeans version

Apache NetBeans 16 release candidate

### What happened

I arrive directly on the second step of the wizard and no Datasource is displayed, so I have to give up. Instead of "Data source", "Please wait.." is displayed, and I waited a very long time. Yet I can connect to the database from the entry "Databases" of the Services tab of NetBeans. 

The server is Payara Server Community 6.2022.1 Alpha 4. This problem does not happen with GlassFish 6.2.5.

### How to reproduce

Create a Maven Web project for Jakarta EE 9.1. 
Launch the wizard "New Entity Classes from Database".
 I arrive directly on the second step of the wizard and no Datasource is displayed so I have to give up. Instead of "Data source", "Please wait.." is displayed, and I waited a very long time. Yet I can connect to the MySQL 8.0.25 database from the entry "Databases" of the Services tab of NetBeans. 

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Windows 11 Family 64-bit, version 21H2

### JDK

11

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

It happens every time.
Related to #3813.

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

### PR DESCRIPTION
**Pull request by asbachb**
Added ability for Jakarta EE projects to:
* Create entities from database tables
* Create message driven, stateful, stateless and session beans

This should fix #4827 #4828

### CODE REVIEW

**Review by matthiasblaesing — COMMENTED** [review]
Currently the license check fails because the new templates have no license header. In that modules the templates are assigned a license using the `licenseinfo.xml` file in the root of the module. Please add the new templates to the list.
I left two further comments inline.

File: enterprise/j2ee.ejbcore/nbproject/org-netbeans-modules-j2ee-ejbcore.sig [hunk 1]
```diff
@@ -159,6 +159,8 @@ meth public java.awt.im.InputMethodRequests getInputMethodRequests()
 meth public java.awt.image.ColorModel getColorModel()
 meth public java.awt.image.VolatileImage createVolatileImage(int,int)
 meth public java.awt.image.VolatileImage createVolatileImage(int,int,java.awt.ImageCapabilities) throws java.awt.AWTException
+meth public java.awt.peer.ComponentPeer getPeer()
```
[the remark this answers is not in the document]
Comment (asbachb) [reply enterprise/j2ee.ejbcore/nbproject/org-netbeans-modules-j2ee-ejbcore.sig]: Removed.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/entity/EntityWizard.java [hunk 2]
```diff
@@ -135,17 +138,26 @@ public void initialize(WizardDescriptor wizardDescriptor) {
     
     @Override
     public Set instantiate() throws IOException {
+        Project project = Templates.getProject(wiz);
+        
+        JPAModuleInfo moduleInfo = project.getLookup().lookup(JPAModuleInfo.class);
+        Double webXmlVersion = Double.parseDouble(moduleInfo.getVersion());
+
+        String enterprisePrefix = "jakarta";
+        if (webXmlVersion < JAKARTA_EE_9_WEB_APP_DD_VERSION) {
```
[the remark this answers is not in the document]
Comment (asbachb) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/entity/EntityWizard.java]: I double checked with an EJB Module project which you're right does not match the logic.

Replaced it with a classpath check as suggested.

**Comment by asbachb** [comment]
Added files to `licenseinfo.xml`.

**Comment by matthiasblaesing** [comment]
If noone objects, I'll merge on the weekend. @juneau001 you might want to have a look.

**Review by juneau001 — COMMENTED** [review]
I have tested this PR and the entity classes do generate.  However, when I generate on a Jakarta EE 9 or Jakarta EE 10 project, the classes generated with javax.* imports.  I tested on existing projects...not new projects.  Thanks for the work on this!

**Comment by asbachb** [comment]
@juneau001 @matthiasblaesing @mbien can we get this back to move somehow? Would be nice to have it in NB19.

I created two maven web applications one with jakarta ee 10 and one with jakarta ee 9 in NetBeans 18 and opened it with the PR. Stateful and stateless session beans were created with the correct imports.
So I'm currently unable to reproduce the mentioned behavior without further details. @juneau001 can you double check?

**Review by mbien — APPROVED** [review]
code looks good to me

**Review by matthiasblaesing — CHANGES_REQUESTED** [review]
I think I see what @juneau001 saw: The title of this PR contains "Jakarta EE Enttiy", so this implies that I can generate @Entity beans. This generates code with wrong package. Steps to reproduce:

Create Java EE 10 Web Project
=============================

- Build this branch, remove `nbbuild/testuserdir`, run `ant tryme`
- Create a new project "Java with Maven" > "Web Application"
- Fill Name and Location Data (not relevant for this test)
- Select "Add" for Server and choose "Glassfish", choose a local folder and download 7.0.5
- Accept domain1 defaults
- Ensure Java EE version is set to "Jakarta EE 10 Web"
- Generate project

Prepare JavaDB Connection
=========================

- Go to services tab
- Select Database -> Drivers -> "Java DB (Network)",  Choose "Customize"
- Add "derbytools.jar" from $glassfishDir/javadb/lib folder

Try to generate entity classes
==============================

- Open Context menu on project node, choose "New" -> "Entity classes from Database"
- In the "Data Source" dropdown select `jdbc/sample`
- Select table "CUSTOMER" and the "Add"
- Select "Next" until to reach the last page of the wizard, then select "Finish"

=> Now check the generated classes, for example "Customer.java". You'll see, that the `javax.persistence` classes are used, although we are in an JakartaEE project

**Comment by mbien** [comment]
can we merge this and fix the remaining issue during RC1?

my understanding is:
 - this PR causes no known regression
 - it improves the current situation
 - the remaining problem is fixable without having to rewrite everything
 
 deadline is today night UTC

**Comment by matthiasblaesing** [comment]
Thanks for the update, I'm doing a local sanity build and if  ok, I'll merge.

**Review by matthiasblaesing — APPROVED** [review]

**Pull request merged**